### ISSUE TITLE
Variance of type argument is incompatible with base class

### ISSUE THREAD

**Issue by LeeeeT** [microsoft/pyright#4818]
Pyright gives me errors for the following code:

```python
from typing import Generic, TypeVar

T_co = TypeVar("T_co", covariant=True)
T_contra = TypeVar("T_contra", contravariant=True)

class Co(Generic[T_co]): ...

class Contra(Generic[T_contra]): ...


class ContraToContra(Contra[Contra[T_co]]): ...

class ContraToCo(Co[Contra[T_contra]]): ...
```

```
file.py:11:29 - error: Type "Contra[T_co@ContraToContra]" cannot be assigned to type variable "T_contra@Contra"
    Variance of type argument "Contra[T_co@ContraToContra]" is incompatible with base class "Contra" (reportGeneralTypeIssues)
file.py:13:21 - error: Type "Contra[T_contra@ContraToCo]" cannot be assigned to type variable "T_co@Co"
    Variance of type argument "Contra[T_contra@ContraToCo]" is incompatible with base class "Co"
```

I expect no errors for this code.
Discussed in https://github.com/microsoft/pyright/issues/4111.

pyright 1.1.300

**Comment by erictraut** [comment]
Thanks for your patience on this issue. It took me a while to understand the necessary logic to get the nested case correct. I used your full test cases and augmented them with a few negative examples (those that should produce errors).

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.301, which I just published. It will also be included in a future release of pylance.
